browser/src/control/jsdialog/Widget.TreeView.ts

```diff
@@ -1235,6 +1143,81 @@ class TreeViewControl {
 
 		return true;
 	}
+
+	// --------- Event Handlers
+	//
+	onClick(e: any) {
+		let target = e.target;
+		let row = TreeViewControl.getElement(target, 'row');
+		if (row) {
+			this.onRowClick(row);
+		}
+
+		if (target.localName === 'input') {
+			this.onCheckBoxClick(target);
+			return;
+		}
+	}
+
+	onRowClick(row: any) {
+		if (L.DomUtil.hasClass(row, 'disabled')) {
+			return;
+		}
+
+		this._container
+			.querySelectorAll('.ui-treeview-entry.selected')
+			.forEach((item: HTMLElement) => {
+				this.unselectEntry(item);
+			});
+
+		this.selectEntry(row);
+
+		this._builder.callback(
+			'treeview',
+			'select',
+			this._data,
+			row._row,
+			this._builder,
+		);
+
+		if (this._singleClickActivate) {
+			this._builder.callback(
+				'treeview',
+				'activate',
+				this._data,
+				row._row,
+				this._builder,
+			);
+		}
+	}
+
+	onCheckBoxClick(checkbox: any) {
+		if (checkbox.checked) {
```

Comment:
what if is disabled? shouldn't do anything then

`if (L.DomUtil.hasClass(row, 'disabled'))`

Reply:
I know, I am still drafting to equivalent code. Thanks for the notice